ott/geometry/costs.py

```diff
@@ -122,27 +124,111 @@ def __init__(self, dimension: int, **kwargs: Any):
     self._dimension = dimension
     self._sqrtm_kw = kwargs
 
-  def norm(self, x: jnp.ndarray) -> Union[jnp.ndarray, float]:
-    norm = jnp.sum(x[..., 0:self._dimension] ** 2, axis=-1)
-    x_mat = jnp.reshape(
-        x[..., self._dimension:], (-1, self._dimension, self._dimension)
-    )
-
-    norm += jnp.trace(x_mat, axis1=-2, axis2=-1)
+  def norm(self, x):
+    if len(x.shape) == 1:
+      num_components = 1
+    else:
+      num_components = x.shape[0]
+    mean, cov = self.x_to_mean_and_cov(x, num_components=num_components)
+    norm = jnp.sum(mean ** 2, axis=-1)
+    norm += jnp.trace(cov, axis1=-2, axis2=-1)
     return norm
 
-  def pairwise(self, x: jnp.ndarray, y: jnp.ndarray) -> float:
-    mean_dot_prod = jnp.vdot(x[0:self._dimension], y[0:self._dimension])
-    x_mat = jnp.reshape(x[self._dimension:], (self._dimension, self._dimension))
-    y_mat = jnp.reshape(y[self._dimension:], (self._dimension, self._dimension))
-
-    sq_x = matrix_square_root.sqrtm(x_mat, self._dimension, **self._sqrtm_kw)[0]
-    sq_x_y_sq_x = jnp.matmul(sq_x, jnp.matmul(y_mat, sq_x))
+  def pairwise(self, x, y):
+    mean_x, cov_x = self.x_to_mean_and_cov(x, num_components=1)
```

Comment:
batch_size?

maybe put in the x_to_mean_and_cov the batch_size = 1 by default to avoid specifying it here.